sparse/_coo/common.py

```diff
@@ -53,17 +53,6 @@ def asCOO(x, name="asCOO", check=True):
     return x
 
 
-def linear_loc(coords, shape):
-    out = np.zeros(coords.shape[1], dtype=np.intp)
-    tmp = np.zeros(coords.shape[1], dtype=np.intp)
-    strides = int(1)
-    for i, d in enumerate(shape[::-1]):
-        np.multiply(coords[-(i + 1), :], strides, out=tmp)
-        np.add(tmp, out, out=out)
-        strides *= d
-    return out
```

Comment:
@hameerabbasi: want me to revise this as:
```python
def ravel_multi_index(coords, shape):
    if shape == ():
        # https://github.com/numpy/numpy/issues/15690
        assert coords.shape[0] == 0
        return np.zeros(coords.shape[1:], dtype=np.intp)
    else:
        return np.ravel_multi_index(coords, shape)
```

Reply:
Yes. I arrived at the same conclusion you did, and you beat me to reporting the bug.

Reply:
Want me to rename it or leave the function name as `linear_loc`?

Reply:
Either is fine with me.